### PR TITLE
slightly improve tests

### PR DESCRIPTION
**Pull request by brad-decker**
I am working on upgrading our keyring libraries to support latest versions of `ethereumjs/tx`, but in so doing i'd like to test it so that I can feel confident in my work. When doing that I noticed a few things:

1. Current tests did not cover the response handlers for signTransaction or signPersonalMessage. All that was being tested is that a listener was attached to the window for the postMessage to the iframe.
2. Tests for the above methods are not idempotent because they used a timeout and immediately returned done() without waiting for the result.

This PR:
1. improves the test setup by using a chai sandbox.
2. improves test setup by adding an account in the root level beforeEach
3. improves test setup by spying on/stubbing out the unlock method in the root level beforeEach
4. improves test setup by restoring sandbox afterEach
5. removes a redundant test
6. makes the signPersonalMessage/signTransaction tests idomptent 
7. slightly improves the reliability of the tests by mocking the ledger integration and returning faked data.

### CODE REVIEW

File: test/test-eth-ledger-bridge-keyring.js [hunk 1]
```diff
@@ -47,10 +49,19 @@ chai.use(spies)
...
 
   let keyring
+  let sandbox
 
-  beforeEach(function () {
+  beforeEach(async function () {
+    sandbox = chai.spy.sandbox()
     keyring = new LedgerBridgeKeyring()
     keyring.hdk = fakeHdKey
+    keyring.setAccountToUnlock(fakeAccounts[0])
```
Comment (Gudahtt) [inline test/test-eth-ledger-bridge-keyring.js:58]: This looks like a mistake. This method is supposed to take the _index_ of the account that you want to be unlocked, and it is initialized at 0 anyway, which is what it seems like you're trying to set it to.

Right now you're passing in the _address_ and it's being parsed as a _decimal_ number, as an index.

File: test/test-eth-ledger-bridge-keyring.js [hunk 2]
```diff
@@ -47,10 +49,19 @@ chai.use(spies)
...
+    sandbox.on(keyring, 'unlock', (_) => Promise.resolve(fakeAccounts[0]))
+    await keyring.addAccounts()
```
Comment (Gudahtt) [inline test/test-eth-ledger-bridge-keyring.js:60]: Nit: Is this step relevant to all of the tests? Generally I've been trying to reduce the amount of stuff done in the `beforeEach`, as it makes it difficult to understand what each individual tests needs.

**Review by Gudahtt — CHANGES_REQUESTED** [review]

File: test/test-eth-ledger-bridge-keyring.js (hunk 1 above)
Comment (brad-decker) [reply test/test-eth-ledger-bridge-keyring.js]: Thanks, fixed in latest.

File: test/test-eth-ledger-bridge-keyring.js (hunk 2 above)
Comment (brad-decker) [reply test/test-eth-ledger-bridge-keyring.js]: It was relevant to about 10 of the thirty tests, so I extracted it into a helper method that can be called in the relevant tests. I then returned the tests that were modified due to this change back to their original state.

File: test/test-eth-ledger-bridge-keyring.js [hunk 3]
```diff
@@ -369,6 +380,8 @@ describe('LedgerBridgeKeyring', function () {
     const accountIndex = 5
     let accounts = []
     beforeEach(async function () {
+      keyring = new LedgerBridgeKeyring()
```
Comment (Gudahtt) [inline test/test-eth-ledger-bridge-keyring.js:383]: Is this now doing the same thing as the top-level beforeEach? :thinking:

File: test/test-eth-ledger-bridge-keyring.js [hunk 4]
```diff
@@ -385,19 +398,6 @@ describe('LedgerBridgeKeyring', function () {
     })
   })
 
-  describe('signMessage', function () {
-    it('should call create a listener waiting for the iframe response', function (done) {
-
-      chai.spy.on(window, 'addEventListener')
-      setTimeout((_) => {
-        keyring.signPersonalMessage(fakeAccounts[0], '0x123')
```
Comment (Gudahtt) [inline test/test-eth-ledger-bridge-keyring.js:393]: :sweat_smile:  Well now.

Technically there is a `signMessage` public method that isn't being tested though. It's a bit silly, in that it's just an alias of `signPersonalMessage`. But, maybe worth duplicating the `signPersonalMessage` test for `signMessage` as well, to ensure the alias is tested and works the same way?

File: test/test-eth-ledger-bridge-keyring.js (hunk 2 above)
Comment (Gudahtt) [reply test/test-eth-ledger-bridge-keyring.js]: The helper function looks great! Much easier to read.

**Review by Gudahtt — COMMENTED** [review]
Looks great! I had a few more smaller suggestions

File: test/test-eth-ledger-bridge-keyring.js [hunk 5]
```diff
@@ -431,60 +431,64 @@ describe('LedgerBridgeKeyring', function () {
...
-      chai.spy.restore(keyring, 'unlock')
-    })
-
-    it('should unlock the given account if found on device', function () {
-      chai.spy.on(keyring, 'unlock', (_) => Promise.resolve(fakeAccounts[0]))
-
-      return keyring.unlockAccountByAddress(fakeAccounts[0])
+    it('should unlock the given account if found on device', async function () {
+      await basicSetupToUnlockOneAccount()
+      keyring.unlockAccountByAddress(fakeAccounts[0])
```
[the remark this answers is not in the document]
Comment (brad-decker) [reply test/test-eth-ledger-bridge-keyring.js]: Done!

File: test/test-eth-ledger-bridge-keyring.js (hunk 3 above)
Comment (brad-decker) [reply test/test-eth-ledger-bridge-keyring.js]: yeah good call, removed

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by mostafaadel33 — APPROVED** [review]

File: test/test-eth-ledger-bridge-keyring.js (hunk 1 above)
Comment (mostafaadel33) [reply test/test-eth-ledger-bridge-keyring.js]: test/test-eth-ledger-bridge-keyring.js

**Review by mehdi167 — APPROVED** [review]

**Review by mehdi167 — APPROVED** [review]

**Review by mehdi167 — APPROVED** [review]
